ts/utils/messages.ts

```diff
@@ -215,21 +246,23 @@ export const isCtaActionValid = (cta: CTA): boolean => {
  * return true if at least one of the CTAs is valid
  * @param ctas
  */
-export const hasCtaValidActions = (ctas: CTAS): boolean => {
-  const isCTA1Valid = isCtaActionValid(ctas.cta_1);
+export const hasCtaValidActions = (
```

Comment:
Could you update the function documentation with the new parameter?

Reply:
Here a more functional refactoring considering we can add more routes with the same logic (i.e regione umbria)

```typescript
type MaybePotMetadata = pot.Pot<ServiceMetadata | undefined, Error> | undefined;
const hasMetadataTokenName = (metadata: MaybePotMetadata): boolean =>
  fromNullable(metadata)
    .map(m =>
      pot.getOrElse(
        pot.map(m, m => m !== undefined && m.token_name !== undefined),
        false
      )
    )
    .getOrElse(false);

// a mapping between routes name (the key) and predicates (the value)
// the predicate says if for that specific route the navigation is allowed
const internalRoutePredicates: Map<
  string,
  Predicate<MaybePotMetadata>
> = new Map<string, Predicate<MaybePotMetadata>>([
  [ROUTES.SERVICE_WEBVIEW, hasMetadataTokenName]
]);

/**
 * return a boolean indicating if the cta action is valid or not
 * @param cta
 */
export const isCtaActionValid = (
  cta: CTA,
  serviceMetadata?: MaybePotMetadata
): boolean => {
  // check if it is an internal navigation
  const maybeInternalRoute = getInternalRoute(cta.action);
  if (maybeInternalRoute.isSome()) {
    return fromNullable(
      internalRoutePredicates.get(maybeInternalRoute.value.routeName)
    )
      .map(f => f(serviceMetadata))
      .getOrElse(true);
  }
  const maybeCustomHandledAction = deriveCustomHandledLink(cta.action);
  // check if it is a custom action (it should be composed in a specific format)
  if (maybeCustomHandledAction.isSome()) {
    return true;
  }
  return false;
};
```

Furthermore could you write some test on this update function?
`ts/utils/__tests__/messages.test.ts`